### PR TITLE
CIRC-1834: Respect allowed service points in Request Policy

### PR DESCRIPTION
**Pull request by OleksandrVidinieiev**
## Purpose
Make use of the new setting (allowed service points) in Request Policy.
Resolves [CIRC-1834](https://issues.folio.org/browse/CIRC-1834).

## Approach
Every time we validate a request against a Request Policy, we should also verify that the requested pickup service point is allowed by this policy.

#### TODOS and Open Questions
- [ ] Check logging

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java [hunk 1]
```diff
@@ -50,13 +52,23 @@ static Result<RequestAndRelatedRecords> refuseWhenRequestCannotBeFulfilled(
...
+    Request request = requestAndRelatedRecords.getRequest();
+    RequestType requestType = request.getRequestType();
 
     if (!requestPolicy.allowsType(requestType)) {
       return failureDisallowedForRequestType(requestType);
-    } else {
-      return succeeded(requestAndRelatedRecords);
     }
+
+    if (!requestPolicy.allowsServicePoint(requestType, request.getPickupServicePointId())) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:62]: Log this if?

File: src/main/java/org/folio/circulation/domain/policy/RequestPolicy.java [hunk 2]
```diff
@@ -60,20 +59,33 @@ public boolean allowsType(RequestType type) {
     return false;
   }
 
-  private static Map<RequestType, Set<UUID>> extractAllowedServicePoints(
+  public boolean allowsServicePoint(RequestType requestType, String servicePointId) {
+    log.debug("allowsPickupServicePoint:: parameters: requestType: {}, servicePointId: {}",
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/policy/RequestPolicy.java:63]: Wrong method name. BTW, shouldn't it be more descriptive? Like, servicePointIsAllowedForRequestType or something similar?

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java [hunk 3]
```diff
@@ -50,13 +52,23 @@ static Result<RequestAndRelatedRecords> refuseWhenRequestCannotBeFulfilled(
...
+      return failedValidation("One or more Pickup Locations are no longer available",
```
Comment (alexanderkurash) [inline src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:63]: I wonder if we should start collecting these messages in one place, maybe we should create a tech debt task for this. We've already been asked to provide all possible validation error messages once.

File: src/main/java/org/folio/circulation/domain/policy/RequestPolicy.java (hunk 2 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/domain/policy/RequestPolicy.java:63]: "servicePointIsAllowed" is pretty much the same as "allowsServicePoint", and request type is specified as one of the parameters so I'm not sure if it makes sense to also mention it in the method name

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/circulation/domain/RequestServiceUtility.java (hunk 3 above)
Comment (OleksandrVidinieiev) [reply src/main/java/org/folio/circulation/domain/RequestServiceUtility.java:63]: Will create a ticket for that

**Review by roman-barannyk — APPROVED** [review]

**Pull request merged**